Source/JavaScriptCore/runtime/ObjectConstructorInlines.h

```diff
@@ -50,7 +50,64 @@ ALWAYS_INLINE bool canPerformFastPropertyNameEnumerationForJSONStringifyWithSide
     return true;
 }
 
-ALWAYS_INLINE bool objectAssignFast(VM& vm, JSObject* target, JSObject* source, Vector<RefPtr<UniquedStringImpl>, 8>& properties, MarkedArgumentBuffer& values)
+ALWAYS_INLINE void objectAssignIndexedPropertiesFast(JSGlobalObject* globalObject, JSObject* target, JSObject* source)
+{
+    VM& vm = globalObject->vm();
```

Comment:
Use `forEachIndexedProperty` to implement `objectAssignIndexedPropertiesFast`.
Stop iteration when exceptions happens with `IterationStatus::Done`.